api/services/StorageService.js

```diff
@@ -92,5 +93,59 @@ module.exports = {
       return true;
     }
     return false;
+  },
+
+  /**
+   * checkUploadLimit
+   *
+   * Check if the upload limit is reached
+   *
+   * @param {Object} Storage of user want to upload
+   * @param {int} Length in byte of the new file
+   * @return {Promise} Error if upload limit is reached
+   */
+  checkUploadLimit: function(storage, length) {
+    return ConfigService.get('uploadLimit')
+      .then((limit) => {
+        return this.storageSize(storage, "", length)
+          .then((size) => {
+            return new Promise(function(resolve, reject) {
+              // limit["uploadLimit"] is in MB
+              if (limit["uploadLimit"] !== 0 && size > limit["uploadLimit"] * 1048576) {
+                return reject({
+                  statusCode: 403,
+                  message: "The upload limit is reached"
+                });
+              }
+              return resolve();
```

Comment:
resolve to a `null` value. `Undefinied` is ambiguous. You can't know if it's an error or an expected empty value.